cmd/api/src/auth/saml.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2023 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/crewjam/saml"
+	"github.com/crewjam/saml/samlsp"
+	dsig "github.com/russellhaering/goxmldsig"
+	"github.com/specterops/bloodhound/crypto"
+	"github.com/specterops/bloodhound/src/config"
+	"github.com/specterops/bloodhound/src/model"
+)
+
+// Todo Verify this is the optimal place for these
```

Comment:
I wasn't exactly sure where the best spot for this was, open to suggestions here!

Reply:
This and the other function below could go inside `cmd/api/src/api/v2/auth/saml.go` since they're both called from that file. But I can understand why you placed them here too!

Reply:
No strong feels here. Keep it for now and if we find this is not very discoverable we can always change it later